src/modules/registrypreview/RegistryPreviewExt/dllmain.cpp

```diff
@@ -0,0 +1,253 @@
+
+    // Return the localized display name of the powertoy
+    virtual const wchar_t* get_name() override
+    {
+        return MODULE_NAME;
+    }
+
+    // Return the non localized key of the powertoy, this will be cached by the runner
+    virtual const wchar_t* get_key() override
+    {
+        return MODULE_NAME;
+    }
+
+    // Return the configured status for the gpo policy for the module
+    virtual powertoys_gpo::gpo_rule_configured_t gpo_policy_enabled_configuration() override
+    {
+        return powertoys_gpo::getConfiguredRegistryPreviewEnabledValue();
+    }
+
+    // Return JSON with the configuration options.
+    virtual bool get_config(wchar_t* buffer, int* buffer_size) override
+    {
+        HINSTANCE hinstance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+
+        // Create a Settings object.
+        PowerToysSettings::Settings settings(hinstance, get_name());
+        settings.set_description(MODULE_DESC);
+
+        return settings.serialize_to_buffer(buffer, buffer_size);
+    }
+
+    // Pop open the app, if the OOBE page asks it to
+    virtual void call_custom_action(const wchar_t* action) override
+    {
+        try
+        {
+            PowerToysSettings::CustomActionObject action_object =
+                PowerToysSettings::CustomActionObject::from_json_string(action);
+
+            if (action_object.get_name() == L"Launch")
+            {
+                launch_process();
+                Trace::ActivateEditor();
+            }
+        }
+        catch (std::exception&)
+        {
+            Logger::error(L"Failed to parse action. {}", action);
+        }
+    }
+
+    // Called by the runner to pass the updated settings values as a serialized JSON.
+    virtual void set_config(const wchar_t* config) override
+    {
+        try
+        {
+            // Parse the input JSON string.
+            PowerToysSettings::PowerToyValues values = PowerToysSettings::PowerToyValues::from_json_string(config, get_key());
+
+            // If you don't need to do any custom processing of the settings, proceed
+            // to persists the values.
+            values.save_to_settings_file();
+        }
+        catch (std::exception&)
+        {
+            // Improper JSON.
+        }
+    }
+
+    // Enable the powertoy
+    virtual void enable()
+    {
+        // Inject the Preview command into the context menu for REG files in Explorer
+        // Puts the values into HKCU to avoid having to elevate privilege
+        WCHAR executable_path[MAX_PATH];
+        ZeroMemory(executable_path, sizeof(executable_path));
+        GetCurrentDirectory(MAX_PATH, executable_path);
+
+        std::wstring command = executable_path;
+        command.append(L"\\modules\\RegistryPreview\\PowerToys.RegistryPreview.exe \"%1\"");
+
+        HKEY key{}, subKey{};
+        auto res = RegOpenKey(HKEY_CURRENT_USER, L"Software\\Classes\\regfile", &key);
+        res = RegCreateKey(key, L"shell\\preview\\command", &subKey);
```

Comment:
@stefansjfw
If we create a Reg_SZ named `icon` under `shell\priview` and set the value to an icon or exe file we can assign an icon to the context menu. what do you think?

Reply:
nice idea. Added as well.
![image](https://user-images.githubusercontent.com/57057282/227282373-12848032-eb46-4a6d-a704-57816fc854db.png)